common/CMakeLists.txt

```diff
@@ -33,11 +37,25 @@ add_library(common_key_value_database STATIC
 )
 target_link_libraries(common_key_value_database PRIVATE platform_compiler_flags)
 target_link_libraries(common_key_value_database INTERFACE crossplatform_compiler_flags)
+if (${MENDER_USE_LMDB} STREQUAL 1)
+  target_sources(common_key_value_database PRIVATE key_value_database/platform/lmdb/lmdb.cpp)
+  target_include_directories(common_key_value_database PRIVATE ${CMAKE_SOURCE_DIR}/vendor/lmdbxx)
+  target_link_libraries(common_key_value_database PUBLIC lmdb)
+endif()
 
 # Test Mender Key Value Database
 add_executable(key_value_database_test EXCLUDE_FROM_ALL key_value_database_test.cpp)
-target_link_libraries(key_value_database_test PRIVATE common_error common_key_value_database GTest::gtest_main gmock)
-gtest_discover_tests(key_value_database_test)
+target_link_libraries(key_value_database_test PRIVATE
+  common_testing
+  common_error
+  common_key_value_database
+  GTest::gtest_main
+  gmock
+)
+target_compile_definitions(key_value_database_test PRIVATE MENDER_USE_LMDB=${MENDER_USE_LMDB})
```

Comment:
Q: What's going on here?

Reply:
We need the define in the code. It's equivalent to `-D MENDER_USE_LMDB=${MENDER_USE_LMDB}`.